src/custom/state/fee/updater.ts

```diff
@@ -1,22 +1,62 @@
-import { useEffect, useRef } from 'react'
+import { useEffect } from 'react'
 import { useActiveWeb3React } from 'hooks'
-import { useAddFee, useAllFees } from './hooks'
 import { useSwapState, tryParseAmount } from 'state/swap/hooks'
 import useIsWindowVisible from 'hooks/useIsWindowVisible'
-import { FeesMap } from './reducer'
-import { getFeeQuote, FeeQuoteParams } from 'utils/operator'
-import { registerOnWindow } from 'utils/misc'
 import { Field } from 'state/swap/actions'
 import { useCurrency } from 'hooks/Tokens'
 import useDebounce from 'hooks/useDebounce'
+import { useAllFees } from './hooks'
+import { useRefetchFeeCallback } from '@src/custom/hooks/useRefetchFee'
+import { FeeQuoteParams } from '@src/custom/utils/operator'
+import { FeeInformationObject } from './reducer'
 
-function isDateLater(dateA: string, dateB: string): boolean {
-  const [parsedDateA, parsedDateB] = [Date.parse(dateA), Date.parse(dateB)]
+const DEBOUNCE_TIME = 350
+const REFETCH_FEE_CHECK_INTERVAL = 15000 // Every 15s
+const RENEW_FEE_QUOTES_BEFORE_EXPIRATION_TIME = 30000 // Will renew the quote if there's less than 30 seconds left for the quote to expire
+
+/**
+ * Returns true if the fee quote expires soon (in less than RENEW_FEE_QUOTES_BEFORE_EXPIRATION_TIME milliseconds)
+ */
+function isExpiringSoon(quoteExpirationIsoDate: string): boolean {
+  const quoteExpirationDate = Date.parse(quoteExpirationIsoDate)
+  // const secondsLeft = (quoteExpirationDate.valueOf() - Date.now()) / 1000
+  // console.log(
+  //   `[state:fee:updater] Fee isExpiring in ${secondsLeft}. Refetch?`,
+  //   quoteExpirationDate <= Date.now() + RENEW_FEE_QUOTES_BEFORE_EXPIRATION_TIME
+  // )
```

Comment:
if it's ok, allow me to leave this dead code, it was useful. It might still be, and it doesn't bother me much as a comment.

Reply:
I wouldn't mind leaving it uncommented